python/revng/api/invalidations.py

```diff
@@ -0,0 +1,30 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+from dataclasses import dataclass
+from typing import Generic, TypeVar
+
+from ._capi import _api
+from .utils import make_python_string
+
+
+class Invalidations:
+    def __init__(self, invalidations=None):
+        if invalidations is None:
+            self._invalidations = _api.rp_invalidations_create()
+        else:
+            self._invalidations = invalidations
+
+    def __str__(self):
+        res = _api.rp_invalidations_serialize(self._invalidations)
+        return make_python_string(res)
```

Comment:
Would be nice to have a list.

Reply:
Eh, rp_invalidations under the hood is implemented as:
```
llvm::StringMap<llvm::StringMap<std::vector<pipeline::Target>>>
```
So having an enumeration is, while not impossible, somewhat tricky.
Moreover I'd argue that it's useless, if you want validations you want them all, so you call `serialize`
Moreover I'd be against serializing on the python side, that would turn a single ffi call in (in a trivial case) a few thousands

Reply:
Mmmh, OK.